.gitignore

```diff
@@ -0,0 +1,139 @@
+
+# Created by https://www.gitignore.io/api/java,macos,gradle,intellij
+
```

Comment:
So, even though you have a .gitignore file, you can tell git is still tracking some files that it shouldn't because they are still turning up here in your repo (like `.idea/misc.xml` and `03-custom-array-list.iml`. This is the command to get git to behave:

`git rm --cached <file1> <file2>`